engine/src/main/java/com/google/android/fhir/impl/FhirEngineImpl.kt

```
@@ -81,7 +81,11 @@ internal class FhirEngineImpl(private val database: Database, private val contex
   }
 
   override suspend fun purge(type: ResourceType, id: String, forcePurge: Boolean) {
-    database.purge(type, id, forcePurge)
+    database.purge(type, setOf(id), forcePurge)
+  }
+
+  override suspend fun purge(type: ResourceType, ids: Set<String>, forcePurge: Boolean) {
+    database.purge(type, ids, forcePurge)
```

Should this fail, I think we should report to the application "why".
Maybe propagate the transaction failure with a proper message ?

---

This can fail for a number of reasons:
1. The Resource in question is not found in the DB 
2. The Resource has ref. entries in the local changes table and FORCE_PURGE is false
3. Some other (system) issue

For the first two we throw the exceptions explicitly for each resource processed

---

After the conversation today we've agreed to wrap the whole batch list processing logic in a transaction. Thus if any item fails it throws an `IllegalStateException`. 

Left [the same comment on ticket #2439](https://github.com/google/android-fhir/issues/2439#issuecomment-2036718710) as well